### PR TITLE
fixed output in error pages

### LINKED ISSUE #27807: Can't download if parent folder contains single quote
### Steps:

1. Create a folder "test'x"
1. Create two subfolders "test'x/a" and "test'x/b"
1. In the web UI, select both "a" and "b" then click "Download" on the header bar on the top right

### Expected:

Download works

### Actual:

Error page.

This works correctly without the quote.

### Versions
ownCloud 10.0.0

Regression introduced by https://github.com/owncloud/core/pull/27716

@Peter-Prochaska please have a look

### PR DESCRIPTION
**Pull request by peterprochaska**
## Description
Escape output in error pages

## Related Issue
fixes https://github.com/owncloud/core/issues/27807

## Motivation and Context
Now downloading files or folder with quotes is possible

## How Has This Been Tested?
Download File with quotes and there is no error page

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/private/legacy/template.php [hunk 1]
```diff
@@ -325,7 +325,7 @@ public static function printErrorPage( $error_msg, $hint = '' ) {
 
 		try {
 			$content = new \OC_Template( '', 'error', 'error', false );
-			$errors = [['error' => $error_msg, 'hint' => $hint]];
+			$errors = [['error' => htmlspecialchars($error_msg, ENT_QUOTES), 'hint' => htmlspecialchars($hint, ENT_QUOTES)]];
```
Comment (DeepDiver1975) [inline lib/private/legacy/template.php]: please use 

```php
\OCP\Util::sanitizeHTML($name)

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

**Comment by PVince81** [comment]
Code looks good. How did you test the error page ?

I tried testing the error page as well but the file name doesn't even appear there... Maybe it did in older versions.

download-error

**Comment by PVince81** [comment]
:+1: if CI passes

**Comment by PVince81** [comment]
@Peter-Prochaska please always link to the backports in the original PR and back. Without this it makes it very difficult to track.

**Comment by PVince81** [comment]
CI stalled, @DeepDiver1975 please restart

**Review by DeepDiver1975 — APPROVED** [review]

**Comment by PVince81** [comment]
regression: https://github.com/owncloud/core/issues/28684

**Pull request merged**